File: Recurrsion first/Cpp/factorial.cpp
#include<bits/stdc++.h>
using namespace std;

int factorial(int n){
    if(n == 1){
        return n;
    }
    return n = n * factorial(n-1);
    
}

int main(){
    int num = 5;
    cout<< factorial(5);
}

File: Recurrsion first/Java/factorial.java
package Java;

public class factorial {
    public static void main(String[] args){
        int num = 5;
        System.out.println(factorial(num));
    }

    static int factorial(int n){
        if(n == 1){
            return n;
        }
        n = n * factorial(n-1);
        return n;
    }

}
